docs/intro-react.md

```diff
@@ -0,0 +1,476 @@
+
+> Notice the double curly braces `{{ }}` surrounding `style`‘s width and height. In JSX, JavaScript values are referenced with `{}`. This is handy if you are passing something other than a string as props, like an array or number (`<Cat food={["fish", "kibble"]} /> age={2}`). However, JS object are **_also_** denoted with curly braces: `{width: 193, height: 110}`. Therefore, to pass a JS object in JSX, you must wrap the object in **another pair** of curly braces: `{{width: 193, height: 110}}`
+
+You can add custom cat pictures to your component by:
+
+- adding an `Image` component to your `Cat` component
+- setting `Image`‘s `source` via `Cat`‘s `props` with `{props.picURI}`
+- then passing `picURI` values in each `Cat`
+
+```SnackPlayer name=Custom%20Pictures
+import React from 'react';
+import { Text, View, Image } from 'react-native';
+
+function Cat(props) {
+  return (
+    <View>
+      <Image source={props.picURI} style={{width: 200, height: 200}}/>
+      <Text>Hello, I am {props.name}!</Text>
+    </View>
+  );
+}
+
+export default function Cafe() {
+  return (
+    <View>
+      <Cat name="Maru" picURI="https://placekitten.com/g/200/200" />
+      <Cat name="Jellylorum" picURI="https://placekitten.com/g/200/200" />
+      <Cat name="Spot" picURI="https://placekitten.com/g/200/200" />
+    </View>
+  );
+}
+```
+
+You can build many things with `props` and the Core Components [`Text`](text), [`Image`](image), and [`View`](view)! But to build something interactive, you’ll need `state`.
+
+## State
+
+Props are set by the parent component and can only be changed by the parent component passing down new values. But for individual components to change according to human interactions, they need to be able to accept and store their own data. Fortunately, React components can have their own `state` you can get and set data from.
+
+<div class="toggler">
+  <ul role="tablist" class="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTabs('syntax', 'functional')">
+      State with Function Components
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTabs('syntax', 'classical')">
+      State with Class Components
+    </li>
+  </ul>
+</div>
+
+<block class="functional syntax" />
+
+You can add state to a component by calling [React’s `useState` Hook](https://reactjs.org/docs/hooks-state.html).
+
+> A Hook is a kind of function that lets you “hook into” React features. For example, `useState` is a Hook that lets you add state to function components. You can learn more about [other kinds of Hooks in the React documentation.](https://reactjs.org/docs/hooks-intro.html)
+
+You can declare a component’s state by calling `useState` inside its function like this:
+
+`const [currentValue, setNewValue] = useState("Initial value.");`
```

Comment:
Make this a full code snippet for highlighting? Also it's easy to miss this happens inside the component. I suggest:

```js
function Something() {
  const [currentValue, setNewValue] = useState("Initial value.");`
  // ...
```

Also we try to follow the convention like `thing / setThing` so that they match 1:1.